### PR TITLE
Do not count dependabot commit

### LINKED ISSUE #11: Don't count bump dependancy by dependabot
Don't count commit bump dependancy by dependabot

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Please be specific when describing the PR. Every detail helps: What are you changing? Why?
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/ps-monitor-module-releases/issues/11
| How to test?      | 
| Possible impacts? | 

**Before** 

Screenshot - 2022-08-24T100742 825

**After**

Screenshot - 2022-08-24T100419 659

### CODE REVIEW

File: src/BranchManager.php [hunk 1]
```diff
@@ -69,7 +71,20 @@ public function getReleaseData($repositoryName)
...
         );
 
-        $openPullRequests = $this->client->api('pull_request')->all('prestashop', $repositoryName, array('state' => 'open', 'base' => $usedBranch));
+        // Do not count dependabot commits since last release.
+        $dependabotCommit = 0;
+        if ($releaseDate != 'NA') {
+            $commits = $this->client->api('repo')->commits()->all('prestashop', $repositoryName, array('sha' => $devBranchUsed, 'since' => $release['created_at']));
+
+            foreach ($commits as $commit) {
+                $pos = strpos($commit['commit']['message'], 'dependabot');
```
Comment (matks) [inline src/BranchManager.php]: You "detect" dependabot commits by checking if a given commit message contains "dependabot" in it? 😅 

It is maybe not the most reliable way to detect it. Isn't it possible to look for the PR author and check if it is dependabot?

Comment (ghost) [reply src/BranchManager.php]: Yes, I couldn't find any other way, so I will use the search api but there is a rate limit, I will add a sleep.

File: generate.php [hunk 2]
```diff
@@ -10,17 +10,18 @@
...
 $client = new \Github\Client();
 $token = $argv[1];
-$client->authenticate($token, null, Github\Client::AUTH_HTTP_TOKEN);
+$client->authenticate($token, null, Github\Client::AUTH_ACCESS_TOKEN);
 
 $branchManager = new \App\PrestaShopModulesReleaseMonitor\BranchManager($client);
 
 function getModules($client): array
 {
+    $excludedRepos = ['gamification', 'ps_emailsmanager', 'welcome'];
```
Comment (matks) [inline generate.php]: Hello @okom3pom I'm so sorry for coming back after all this time. I've got some time now :)

PR seems OK but I don't understand why you exclude these 3 repositories. Can you tell me?

Comment (ghost) [reply generate.php]: Hello, these 3 repositories are from PrestaShop Corp.

Comment (matks) [reply generate.php]: Welcome module is on the OSS project organization https://github.com/PrestaShop/welcome

Comment (ghost) [reply generate.php]: Hmm, that's strange. This module is not used anymore in PrestaShop 8.0, I correct the PR

Comment (matks) [reply generate.php]: Autoupgrade module is also not embedded in the software ;) but available on download

File: generate.php [hunk 3]
```diff
@@ -10,17 +10,20 @@
...
 $token = $argv[1];
-$client->authenticate($token, null, Github\Client::AUTH_HTTP_TOKEN);
+$client->authenticate($token, null, Github\Client::AUTH_ACCESS_TOKEN);
 
 $branchManager = new \App\PrestaShopModulesReleaseMonitor\BranchManager($client);
 
 function getModules($client): array
 {
-    $contents = $client->api('repo')->contents()->show('PrestaShop', 'PrestaShop-modules');
+    // PrestaShop Corp reporepositories
```
Comment (matks) [inline generate.php]:
```suggestion
    // PrestaShop Corp repositories
```

**Review by matks — APPROVED** [review]
Thank you @okom3pom

**Comment by matks** [comment]
Argh

```
PHP Fatal error:  Uncaught Github\Exception\RuntimeException: You have exceeded a secondary rate limit. Please wait a few minutes before you try again. in /home/runner/work/ps-monitor-module-releases/ps-monitor-module-releases/vendor/knplabs/github-api/lib/Github/HttpClient/Plugin/GithubExceptionThrower.php:126
```

**Comment by matks** [comment]
@okom3pom I had to revert the PR because of the API calls limit 😭

**Comment by ghost** [comment]
Hum ... i added a sleep for that, it's work in local :pensive: 

```
        // Do not count dependabot commits since last release
        // sleep() for rate limit see : https://docs.github.com/en/rest/search#rate-limit
```

**Comment by matks** [comment]
Maybe GitHub provided token has less calls available that a personal token?

**Pull request merged**